include/ccf/js/registry.h

```diff
@@ -127,6 +131,28 @@ namespace ccf::js
     ccf::ApiResult get_js_runtime_options_v1(
       ccf::JSRuntimeOptions& options, kv::ReadOnlyTx& tx);
 
+    /**
+     * Record action details by storing them in KV maps using a common format,
+     * for the purposes of offline audit using the ledger.
+     */
+    ccf::ApiResult record_action_details_for_audit_v1(
```

Comment:
Not sure about the value of "trail" here, since this is a transactional, point-in-time write, but agree "details" does not add much. `record_action_for_audit_v1` seems just as good.